view.c

```diff
@@ -138,13 +138,19 @@ view_is_transient_for(struct cg_view *child, struct cg_view *parent) {
 void
 view_damage_surface(struct cg_view *view)
 {
-	output_damage_view_surface(view->server->output, view);
+	struct cg_output *output;
+	wl_list_for_each(output, &view->server->outputs, link) {
+		output_damage_view_surface(output, view);
```

Comment:
Can we find out what output the damage occurs on and only call `output_damage_view_surface` on that output?

Reply:
A surface can be on any amount of outputs. In fact, top levels will now by default span all outputs.

The implementation in sway was used as inspiration, which indeed iterates over all outputs (done by `desktop_damage_surface`).

Reply:
Thanks :)